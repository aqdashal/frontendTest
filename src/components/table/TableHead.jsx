import { useState } from "react";
import ArrowUpImage from '../../images/up_arrow.png';
import ArrowDownImage from '../../images/down_arrow.png';
import defaultArrowUpImage from '../../images/default.png';
const icons = {
  "up": ArrowUpImage,
  "down": ArrowDownImage,
  "default": defaultArrowUpImage
}
const TableHead = ({ columns, handleSorting, sortField, sortOrder }) => {

  const handleSortingChange = (accessor) => {
    const order =
      accessor === sortField && sortOrder === "asc" ? "desc" : "asc";
    handleSorting(accessor, order);
  };

  return (
    <thead>
      <tr>
        {columns.map(({ label, accessor, sortable }) => {
          const cl = sortable 
            ? sortField === accessor && sortOrder === "asc"
              ? "up"
              : sortField === accessor && sortOrder === "desc"
                ? "down"
                : "default"
            : "";
          return (
            <th
              key={accessor}
              onClick={sortable ? () => handleSortingChange(accessor) : null}
              className={`${cl} border-bottom`}
            >
              {label}
              <span><img src={icons[cl]}></img></span>
            </th>
          );
        })}
      </tr>
    </thead >
  );
};

export default TableHead;