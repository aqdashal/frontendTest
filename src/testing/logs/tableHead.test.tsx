import React from 'react';
import { render, screen } from "@testing-library/react";
import renderer from 'react-test-renderer';
import { columns } from "../data"
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import TableHead from "../../components/table/TableHead";

describe('<TableHead/>', () => {

    let handleSorting: jest.Mock;
    let sortOrder;
    let sortField;
    let tableColumns;

    const setup = () =>
        render(
            <MemoryRouter>
                <TableHead {...{ columns: tableColumns, sortField, sortOrder, handleSorting }} />
            </MemoryRouter>
        )

    beforeEach(() => {
        sortOrder = "";
        sortField = "";
        tableColumns = columns;
        handleSorting = jest.fn();
    })

    it("it should render initially", () => {
        setup();
    })

    it("it renders TableHead properly", () => {
        setup();

        expect(screen.getByText(/Log ID/i))
        expect(screen.getByText(/Application Type/i))
        expect(screen.getByText(/Application ID/i))
        expect(screen.getByText(/Action Details/i))
        expect(screen.getByText(/Date:Time/i))
    })

    test("snapshot", () => {
        const tree = renderer.create(
            <MemoryRouter>
                <TableHead {...{ columns: tableColumns, sortField, sortOrder, handleSorting }} />
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })

})
