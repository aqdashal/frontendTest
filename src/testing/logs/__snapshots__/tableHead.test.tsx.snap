// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableHead/> snapshot 1`] = `
<thead>
  <tr>
    <th
      className="default border-bottom"
      onClick={[Function]}
    >
      Log ID
      <span>
        <img
          src="default.png"
        />
      </span>
    </th>
    <th
      className="default border-bottom"
      onClick={[Function]}
    >
      Application Type
      <span>
        <img
          src="default.png"
        />
      </span>
    </th>
    <th
      className="default border-bottom"
      onClick={[Function]}
    >
      Application ID
      <span>
        <img
          src="default.png"
        />
      </span>
    </th>
    <th
      className="default border-bottom"
      onClick={[Function]}
    >
      Action
      <span>
        <img
          src="default.png"
        />
      </span>
    </th>
    <th
      className=" border-bottom"
      onClick={null}
    >
      Action Details
      <span>
        <img />
      </span>
    </th>
    <th
      className="default border-bottom"
      onClick={[Function]}
    >
      Date:Time
      <span>
        <img
          src="default.png"
        />
      </span>
    </th>
  </tr>
</thead>
`;
